specification/monitor/resource-manager/Microsoft.Insights/stable/2024-03-11/dataCollectionRules_API.json

```diff
@@ -0,0 +1,2379 @@
+          "example": "/subscriptions/703362b3-f278-4e4b-9179-c76eaf41ffc2/resourceGroups/myResourceGroup/providers/Microsoft.Monitor/accounts/centralTeamAccount"
+        },
+        "accountId": {
+          "description": "The immutable ID of the account.",
+          "type": "string",
+          "readOnly": true,
+          "example": "a6a097e1-15e9-42bc-8f48-fcddd804d0f3",
+          "x-ms-mutability": [
+            "read"
+          ]
+        },
+        "name": {
+          "description": "A friendly name for the destination. \r\nThis name should be unique across all destinations (regardless of type) within the data collection rule.",
+          "type": "string",
+          "example": "myDest1"
+        }
+      }
+    },
+    "OtelDataSourceResourceAttributeRouting": {
+      "description": "Enables OTLP (logs, traces, and metrics) payload routing",
+      "type": "object",
+      "properties": {
+        "attributeName": {
+          "description": "The name of the resource attribute to match.",
+          "type": "string",
+          "example": "service.name"
+        },
+        "attributeValue": {
+          "description": "The value of the resource attribute to match.",
+          "type": "string",
+          "example": "myServiceName"
+        }
+      }
+    },
+    "OtelLogsDataSource": {
+      "description": "Enables Otel logs to be collected by this data collection rule.",
+      "required": [
+        "streams"
+      ],
+      "type": "object",
+      "properties": {
+        "streams": {
+          "description": "List of streams that this data source will be sent to.",
+          "type": "array",
+          "items": {
+            "enum": [
+              "Microsoft-OTel-Logs"
+            ],
+            "type": "string",
+            "x-ms-enum": {
+              "name": "KnownOtelLogsDataSourceStreams",
+              "modelAsString": true
+            }
+          },
+          "example": [
+            "Microsoft-OtelLogs"
+          ]
+        },
+        "resourceAttributeRouting": {
+          "description": "Specifies the routing policy based on OTLP payload resource attributes to route subset of the payload according to matching resource attribute.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/OtelDataSourceResourceAttributeRouting"
+            }
+          ]
+        },
+        "enrichWithResourceAttributes": {
+          "description": "Specifies the list of resource attributes that need to be added as labels/dimensions to the log record for further enrichment.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "example": [
+            "service.name"
+          ]
+        },
+        "enrichWithReference": {
+          "description": "Specifies the reference to enrich the telemetry signal with.",
+          "type": "string",
+          "example": "myAI"
+        },
+        "replaceResourceIdWithReference": {
```

Comment:
>replaceResourceIdWithReference [](http://example.com/codeflow?start=9&length=30)

Enum is recommended over boolean for many reasons. This property could be something more like:
enrichmentType: { UseDefaultResourceId, UseReferencedResourceId }. Others could be added in the future.